systemservices/marketplace/mesg.yml

```diff
@@ -0,0 +1,320 @@
+                    type: String
+                  active:
+                    name: ""
+                    description: ""
+                    type: Boolean
+                  createTime: *createTime
+              purchases: &purchases
+                name: ""
+                description: ""
+                type: Object
+                repeated: true
+                optional: true
+                object: &purchase
+                  purchaser: &purchaser
+                    name: ""
+                    description: ""
+                    type: String
+                  expire: &expire
+                    name: ""
+                    description: ""
+                    type: String
+                  createTime: *createTime
+      error: &outputError
+        name: "Error"
+        description: "Output when an error occurs"
+        data:
+          message:
+            name: "Message"
+            description: "The error message"
+            type: String
+          code:
+            name: "Error code"
+            description: "The error code"
+            type: String
+            optional: true
+  getService:
+    name: ""
+    description: ""
+    inputs:
+      sid: *sid
+    outputs:
+      success:
+        name: ""
+        description: ""
+        data: *service
+      error: *outputError
+  publishServiceVersion:
+    name: ""
+    description: ""
+    inputs:
+      sid: *sid
+      manifest: *manifest
+      manifestProtocol: *manifestProtocol
+      from: &inputFrom
+        name: ""
+        description: ""
+        type: String
+      gas: &inputGas
+        name: ""
+        description: ""
+        type: Number
+        optional: true
+      gasPrice: &inputGasPrice
+        name: ""
+        description: ""
+        type: String
+        optional: true
+    outputs:
+      error: *outputError
+      success: &outputTransaction
+        data: &transaction
+          chainID:
+            name: "Chain ID"
+            description: "The ID of the chain the transaction is for. See https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md#list-of-chain-ids."
+            type: Number
+          nonce:
+            name: "Nonce"
+            description: "Nonce to use."
+            type: Number
+          to:
+            name: "To"
+            description: "The to address."
+            type: String
+          value:
+            name: "Value"
+            description: "The value in Wei."
+            type: String
+          gas:
+            name: "Gas"
+            description: "The maximum gas to use to execute this transaction"
+            type: Number
+          gasPrice:
+            name: "Gas price"
+            description: "The gas price in Wei."
+            type: String
+          data:
+            name: "Data"
+            description: "The data of the transaction."
+            type: String
+  createServiceOffer:
+    name: ""
+    description: ""
+    inputs:
+      sid: *sid
+      price:
```

Comment:
any reason why price is in string ? what's the format ? why not number ?
Same for duration
It makes more sense to have number here, the contract definition accepts number and string and I think we will not use bignumber for duration or price.
Also could be good to add some details on the duration is it seconds / hours / months... maybe having `duration` + `unit`.